### ISSUE TITLE
Unexpected speed up of generated voice conversion samples

### ISSUE THREAD

**Issue by SolomidHero** [yistLin/FragmentVC#14]
Hi! Thank you for your awesome work! I am experiencing some problems when generated voice conversions have much faster speed than source.

In my experiments:
- I use [VCC2020 database](https://github.com/nii-yamagishilab/VCC2020-database) where I am trying conversion of `vcc2020_database_evaluation/[SEM1, SEF1, SEM2, SEF2]` (as source) to `vcc2020_database_groundtruth/TEM1` (as target). 
- For target was used 3-4 of other utterances.
- Processing file: `convert_batch.py`.
- All checkpoints are from pretrained releases. 

Generated speech has great quality but faster speed (E30015.wav utterance was used):
```
source SEM1, 1.975s
SEM1 -> TEM1, 1.012s

source SEM2, 1.343s
SEM2 -> TEM1, 0.7875s

source SEF1, 1.437s
SEF1 -> TEM1, 0.8s

source SEF2, 1.348s
SEF2 -> TEM1, 0.8s
```

In these samples trimming wouldn't give such drop in length, and visually it seems audio to be time stretched.

I also experience same problem with my own audio samples. I tried to resample audios to 16kHz, but effect was here anyway. I wonder if there is quite simple way of removing this unexpected speed up.

Thank you for your help

**Comment by SolomidHero** [comment]
Oh, it seems I shouldn't use checkpoint of vocoder from universal-vocoder repo release since it is the part which produces shortened audio. So I moved to use checkpoint `vocoder.pt` from this repo.

Now I experience bad quality of generating audio of the same data (I even tried same utterance as src and tgt). So, on which data was vocoder trained?

Also I wonder why does `log_mel_spectrogram` uses 326 `hop_len` since wav2vec2 uses about 320 hop length? Is there some  time domain inconsistency in creating mel spectrograms?

**Comment by yistLin** [comment]
Sorry for my late reply!

The vocoder released within this repo was only trained on LibriTTS train-clean-100 and CMU Arctic Dataset, and I guess that's why you encountered generating audio of bad quality.

And the reason why I set hop length to be 326 is because we believed / thought Wav2Vec2 also used this setting. Maybe I didn't understand the original paper correctly? I think we can discuss this furthermore.

**Comment by SolomidHero** [comment]
> And the reason why I set hop length to be 326 is because we believed / thought Wav2Vec2 also used this setting. Maybe I didn't understand the original paper correctly? I think we can discuss this furthermore.

@yistLin yes, it is not correct. To be more precise, wav2vec extract with 320 stride and 400 window. This behaviour corresponds to `n_fft=400, hop_length=320, center=False` parameters of `librosa.stft`.

Also I was working on this model a lot, so could we contact to discuss the improvements which can be made?

**Comment by yistLin** [comment]
Thank you for such information! And yes, I think we can discuss more about how to make some improvements. Maybe you can contact me through email.

**Comment by GreatDarrenSun** [comment]
Have you solved it yet? I also encountered the same problem

**Comment by yistLin** [comment]
@GreatDarrenSun If you encountered unexpected speed-up of synthetic audio, I guess you’re not using the `vocoder.pt` provided in this repo. You can download that from the **Releases** page.

**Comment by GreatDarrenSun** [comment]
Thank you very much.I'm very happy to have received your response.I tried it, and now the speed of synthesized sound is normal.  I want to ask another question, what is the loss of fragment.pt in the release version

**Comment by yistLin** [comment]
It's around 0.39.
